### PR TITLE
Fix current build

### PR DESCRIPTION
**Pull request by roschaefer**
The `master` branch on our frontend is currently failing. It's because we don't set `disabled` and `deleted` to false unless provided. This was a bug introduced in #192.

This PR not only includes the bugfix but also ensures that only authors of post can delete and update their own posts. CC @mattwr18 @mattwr18 @grenzfrequence

### CODE REVIEW

**Comment by appinteractive** [comment]
Why you don’t use the permission shield for permission handling?

**Comment by roschaefer** [comment]
@appinteractive `parent` is `null` because `CreatePost`, and `UpdatePost` are called before updating the post. Which makes sense (obviously) but is also the explanation why we cannot access the post in advance.

**Comment by roschaefer** [comment]
@appinteractive if you want, I can move `isAuthor` from the resolver to `permissionMiddleware` and break our best practice to put `cypher` queries as close to the resolver as possible.

**Comment by roschaefer** [comment]
@appinteractive actually, moving `isAuthor` to permissionsMiddleware seems the better solution to me right now. I'll refactor.

File: src/middleware/softDeleteMiddleware.js [hunk 1]
```diff
@@ -19,5 +19,8 @@ export default {
     User: async (resolve, root, args, context, info) => {
       return resolve(root, setDefaults(args), context, info)
     }
+  },
+  Mutation: async (resolve, root, args, context, info) => {
+    return resolve(root, setDefaults(args), context, info)
```
Comment (appinteractive) [inline src/middleware/softDeleteMiddleware.js]: You set the defaults on ALL mutations? Seems wrong doesn’t it? @roschaefer

Comment (roschaefer) [reply src/middleware/softDeleteMiddleware.js]: No, that's intended. It doesn't hurt to set `deleted: false` and `disabled: false` unless it's provided.

**Pull request merged**